model_analyzer/record/record_aggregator.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+from model_analyzer.record.record import Record
+from collections import defaultdict
+
+
+class RecordAggregator:
+    """
+    Stores a collection of Record objects.
+    """
+    def __init__(self):
+        self._records = defaultdict(list)
+
+    def insert_record(self, record):
+        """
+        Insert a record into the RecordAggregator
+
+        Parameters
+        ----------
+        record : Record
+            A record to be inserted
+        """
+        if isinstance(record, Record):
+            header = record.header()
+            self._records[header].append(record)
+        else:
+            raise TritonModelAnalyzerException(
+                "Can only add objects of type 'Record' to RecordAggregator")
+
+    def get_records(self, headers=None, filters=None):
+        """
+        Get records that satisfy
+        the given list of criteria.
+
+        Parameters
+        ----------
+
+        headers : list of str
+            the names of the records we are
+            imposing the filter criteria on.
+
+        filters : list of callables
```

Comment:
Nice! I think it would great if we mention in this docstring that the filters should have the same size as the headers or should not be provided.